### PR TITLE
CASSANDRA-16398: add ACCESS and DATACENTERS to unreserved keywords

### PR DESCRIPTION
**Pull request by yifan-c**

### CODE REVIEW

File: test/unit/org/apache/cassandra/tools/cqlsh/CqlshTest.java [hunk 1]
```diff
@@ -44,4 +48,29 @@ public void testKeyspaceRequired()
...
+    @Test
+    public void testUseUnreservedKeywordAsColumnName()
+    {
+        List<String> names = Arrays.asList("access", "datacenters");
+        for (String name : names)
+        {
+            testUseUnreservedKeywordAsColumnName(name);
+            testUseUnreservedKeywordAsColumnName(name.toUpperCase());
+        }
+    }
```
Comment (blerer) [inline test/unit/org/apache/cassandra/tools/cqlsh/CqlshTest.java]: nit: I would have put that test into  `CreateTest`. I would not think of looking into CqlshTest for a test on unreserved Keywords.

**Comment by yifan-c** [comment]
@blerer thanks for the feedback. 
I move the test to `CreateTest` and rewrite using `createTable()` instead of invoking cqlsh. 

I also made an optional commit c81be6f that determines all the unreserved keywords and enumerate through to give the best coverage. It helps to prevent the scenario that adds a keyword but forgot to add it to either unreserved or reserved list. 
However, the commit opens up the access level to `ReservedKeywords` class. Would it be a concern? Intuitively, I think "no". Or if we want to keep the access level, we can move the test into `ReservedKeywordsTest` (and potentially change the test class name)?

**Comment by blerer** [comment]
Your new test is a great idea. :-) We often messed up with reserved and unreserved keywords.
I do not believe that changing the access level is an issue.

File: src/java/org/apache/cassandra/cql3/ReservedKeywords.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableSet;
 
-class ReservedKeywords
+public class ReservedKeywords
```
Comment (blerer) [inline src/java/org/apache/cassandra/cql3/ReservedKeywords.java]: We should probably make the class `final`.

File: test/unit/org/apache/cassandra/cql3/validation/operations/CreateTest.java [hunk 3]
```diff
@@ -688,12 +692,16 @@ public void testCreateTableWithCompression() throws Throwable
     @Test
     public void testUseUnreservedKeywordAsColumnName()
     {
-        List<String> names = Arrays.asList("access", "datacenters");
-        for (String colName : names)
+        Set<String> allKeywords = Arrays.stream(CqlLexer.class.getDeclaredFields())
+                                        .filter(f -> f.getName().startsWith("K_"))
+                                        .map(f -> f.getName().substring(2)) // remove the heading "K_"
+                                        .collect(Collectors.toSet());
+        Set<String> unreservedKeywords = Sets.difference(allKeywords, ReservedKeywords.reservedSet);
```
Comment (blerer) [inline test/unit/org/apache/cassandra/cql3/validation/operations/CreateTest.java]: Another approach would be to make `isReserved(String text)` `public` and to use it to filter the stream.
```

Set<String> unreservedKeywords = Arrays.stream(CqlLexer.class.getDeclaredFields())
                                                                  .filter(f -> f.getName().startsWith("K_"))
                                                                  .map(f -> f.getName().substring(2)) // remove the heading "K_"
                                                                  .filter(f -> !ReservedKeywords.isReserved(f))
                                                                 .collect(Collectors.toSet());
```
Making `isReserved(String text)` `public` allow us to expose less of the class internals but I do not really have a strong opinion on which solution we should use.

**Review by blerer — APPROVED** [review]

**Pull request closed without merging**